src/common/interop/HotkeyManager.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+#include <Windows.h>
+#include "KeyboardHook.h"
+
+namespace interop
+{
+public
+    ref struct Hotkey
+    {
+        bool Win;
+        bool Ctrl;
+        bool Shift;
+        bool Alt;
+        unsigned char Key;
+
+        Hotkey()
+        {
+            Win = false;
+            Ctrl = false;
+            Shift = false;
+            Alt = false;
+            Key = 0;
+        }
+    };
+
+public
+    delegate void HotkeyCallback();
+
+typedef unsigned short HOTKEY_HANDLE;
+
+public
+    ref class HotkeyManager
+    {
+    public:
+        HotkeyManager();
+        ~HotkeyManager();
+
+        HOTKEY_HANDLE RegisterHotkey(Hotkey ^ hotkey, HotkeyCallback ^ callback);
```

Comment:
The ^ character means a CLR pointer. This is because this is not standard C++ but rather C++/CLI. It just means a reference to a garbage collected (managed) object.